src/lib/es2015.core.d.ts

```diff
@@ -20,50 +30,50 @@ interface Array<T> {
      * @param thisArg If provided, it will be used as the this value for each invocation of
      * predicate. If it is not provided, undefined is used instead.
      */
-    findIndex(predicate: (value: T, index: number, obj: T[]) => unknown, thisArg?: any): number;
+    findIndex(predicate: (value: T, index: number, array: T[]) => unknown, thisArg?: any): number;
 
     /**
      * Changes all array elements from `start` to `end` index to a static `value` and returns the modified array
      * @param value value to fill array section with
      * @param start index to start filling the array at. If start is negative, it is treated as
-     * length+start where length is the length of the array.
+     * length + start where length is the length of the array.
      * @param end index to stop filling the array at. If end is negative, it is treated as
-     * length+end.
+     * length + end.
      */
-    fill(value: T, start?: number, end?: number): this;
+    fill(value: T, start?: number, end?: number): T[];
 
     /**
      * Returns the this object after copying a section of the array identified by start and end
      * to the same array starting at position target
-     * @param target If target is negative, it is treated as length+target where length is the
+     * @param target If target is negative, it is treated as length + target where length is the
      * length of the array.
-     * @param start If start is negative, it is treated as length+start. If end is negative, it
-     * is treated as length+end.
+     * @param start If start is negative, it is treated as length + start. If end is negative, it
+     * is treated as length + end.
      * @param end If not specified, length of the this object is used as its default value.
      */
-    copyWithin(target: number, start: number, end?: number): this;
+    copyWithin(target: number, start: number, end?: number): T[];
 }
 
 interface ArrayConstructor {
     /**
-     * Creates an array from an array-like object.
-     * @param arrayLike An array-like object to convert to an array.
+     * Creates an array from an array-like or iterable object.
+     * @param source An array-like or iterable object to convert to an array.
      */
-    from<T>(arrayLike: ArrayLike<T>): T[];
+    from<T>(source: ArrayLike<T>): T[];
 
     /**
-     * Creates an array from an iterable object.
-     * @param arrayLike An array-like object to convert to an array.
+     * Creates an array from an array-like or iterable object.
+     * @param source An array-like or iterable object to convert to an array.

```

Comment:
This doesn't look necessary? `from<T>(iterable: Iterable<T> | ArrayLike<T>): T[];` is defined in `es2015.iterable`, and if you include that, passing an iterable would fall to that overload anyway.

Reply:
As stated in #49773, this PR aims to make the text of all the documentations consistent because it may mislead people who are new to JS if they don't look at the doc of the other overloads.
Currently, some are missing "array-like" and some are missing "iterable" despite the location.
Removing all the "iterable" prior to ES2015 will be a big change either.

Reply:
The doc always describes that particular overload. If the overload only accepts an array-like, it doesn't make sense to talk about an iterable. The purpose of JSDoc (IMO) is an after-the-fact explanation, and assumes you know what you are doing in the first place. It's not meant to be educative and cover semantics not included in the type signature.

Reply:
I’ve just combined each pair of overloads into one.